Micronaut Data PostgresSql R2DBC GraalVM native image can't connect to a database

### Expected Behavior

Expected everything to work same as the JVM version

### Actual Behaviour

The application doesn't connect to PostgresSql when packaged natively into a container(dockerBuildNative).

[reactor-tcp-nio-1] ERROR i.m.http.server.RouteExecutor - Unexpected error occurred: Cannot connect to xxx.xx.xx.xxx/<unresolved>:5432
  io.r2dbc.postgresql.PostgresqlConnectionFactory$PostgresConnectionException: Cannot connect to xxx.xx.xx.xxx/<unresolved>:5432
       at io.r2dbc.postgresql.PostgresqlConnectionFactory.cannotConnect(PostgresqlConnectionFactory.java:216)
       at reactor.core.publisher.Mono.lambda$onErrorMap$31(Mono.java:3733)
       at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:94)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onError(ReactorSubscriber.java:64)
       at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:129)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
       at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
       at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
       at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:122)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
       at reactor.core.publisher.MonoDelayUntil$DelayUntilCoordinator.complete(MonoDelayUntil.java:418)
       at reactor.core.publisher.MonoDelayUntil$DelayUntilTrigger.onComplete(MonoDelayUntil.java:531)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onComplete(ReactorSubscriber.java:71)
       at reactor.core.publisher.FluxHandle$HandleSubscriber.onComplete(FluxHandle.java:220)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onComplete(ReactorSubscriber.java:71)
       at reactor.core.publisher.FluxHandle$HandleSubscriber.onComplete(FluxHandle.java:220)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onComplete(ReactorSubscriber.java:71)
       at reactor.core.publisher.FluxCreate$BaseSink.complete(FluxCreate.java:460)
       at reactor.core.publisher.FluxCreate$BufferAsyncSink.drain(FluxCreate.java:805)
       at reactor.core.publisher.FluxCreate$BufferAsyncSink.complete(FluxCreate.java:753)
       at reactor.core.publisher.FluxCreate$SerializedFluxSink.drainLoop(FluxCreate.java:247)
       at reactor.core.publisher.FluxCreate$SerializedFluxSink.drain(FluxCreate.java:213)
       at reactor.core.publisher.FluxCreate$SerializedFluxSink.complete(FluxCreate.java:204)
       at io.r2dbc.postgresql.client.ReactorNettyClient$Conversation.complete(ReactorNettyClient.java:619)
       at io.r2dbc.postgresql.client.ReactorNettyClient$BackendMessageSubscriber.emit(ReactorNettyClient.java:885)
       at io.r2dbc.postgresql.client.ReactorNettyClient$BackendMessageSubscriber.onNext(ReactorNettyClient.java:761)
       at io.r2dbc.postgresql.client.ReactorNettyClient$BackendMessageSubscriber.onNext(ReactorNettyClient.java:667)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
       at reactor.core.publisher.FluxHandle$HandleSubscriber.onNext(FluxHandle.java:126)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
       at reactor.core.publisher.FluxPeek$PeekSubscriber.onNext(FluxPeek.java:200)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
       at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:122)
       at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
       at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:122)
       at reactor.netty.channel.FluxReceive.drainReceiver(FluxReceive.java:279)
       at reactor.netty.channel.FluxReceive.onInboundNext(FluxReceive.java:388)
       at reactor.netty.channel.ChannelOperations.onInboundNext(ChannelOperations.java:404)
       at reactor.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:93)
       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
       at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
       at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:327)
       at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:299)
       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
       at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
       at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
       at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
       at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
       at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
       at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
       at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722)
       at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658)
       at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584)
       at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)
       at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:995)
       at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
       at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
       at java.lang.Thread.run(Thread.java:833)
       at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:704)
       at com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:202)
  Caused by: java.lang.IllegalArgumentException: Class java.time.Instant[] is instantiated reflectively but was never registered.Register the class by adding "unsafeAllocated" for the class in reflect-config.json.
       at com.oracle.svm.core.graal.snippets.SubstrateAllocationSnippets.arrayHubErrorStub(SubstrateAllocationSnippets.java:252)
       at java.util.Collections$UnmodifiableCollection$1.next(Collections.java:1054)
       at io.r2dbc.postgresql.codec.CachedCodecLookup.afterCodecAdded(CachedCodecLookup.java:70)
       at io.r2dbc.postgresql.codec.DefaultCodecs.<init>(DefaultCodecs.java:82)
       at io.r2dbc.postgresql.codec.DefaultCodecs.<init>(DefaultCodecs.java:66)
       at io.r2dbc.postgresql.PostgresqlConnectionFactory.lambda$doCreateConnection$13(PostgresqlConnectionFactory.java:157)
       at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:125)

################

The exact same application connects to a database when packaged into a jar and ran inside a container(dockerBuild).

### Steps To Reproduce

Micronaut Data R2DBC PostgresSql
and use dockerBuildNative to package the application. Fails on the first request to the application

### Environment Information

OS: MacOS Monterey.
JDK: 17.0.4
Docker: 4.10.1

### Example Application

_No response_

### Version

3.5.3

--- a/src/main/docs/guide/availableDrivers.adoc
+++ b/src/main/docs/guide/availableDrivers.adoc
@@ -36,6 +36,8 @@ R2DBC Driver:
 
 dependency:org.postgresql:r2dbc-postgresql[scope="runtimeOnly"]
 
+NOTE: For GraalVM native images, use a recent `org.postgresql:r2dbc-postgresql` version. Older `0.9.x` releases do not include the native-image reflection metadata required by the PostgreSQL codec layer and can fail at runtime with errors about array types such as `Instant[]`. The Micronaut R2DBC BOM manages a compatible version for you, but if you override the driver version manually, use `1.0.2.RELEASE` or newer.
+
 And for Flyway migrations the JDBC driver:
 
 dependency:org.postgresql:postgresql[scope="runtimeOnly"]

